internal/helm/chartyaml_test.go

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package helm
+
+import "testing"
+
+func TestShouldPassIfCanReadChartYaml(t *testing.T) {
+	c := chartYamlFromFile("test/TestChartValid.yaml")
+	if c.Name != "test-application" {
+		t.Errorf("Test should pass but name of the helm chart is not 'test-application'.")
+	}
+}
+
+func TestShouldPassIfChartVersionIsValid(t *testing.T) {
+	c := newChartYaml()
+	c.Version = "1.2.3"
+	if !c.isVersionValid() {
+		t.Errorf("Test should pass since version is valid according Semantic Version schema.")
+	}
+}
+
+func TestShouldFailIfChartVersionIsInvalid(t *testing.T) {
+	c := newChartYaml()
+	c.Version = "1.2.ABC"
+	if c.isVersionValid() {
+		t.Errorf("Test should fail since version is invalid according to Semantic Version schema.")
+	}
+}
+
+func TestShouldPassIfMandatoryFieldsPresent(t *testing.T) {
+	c := chartYamlFromFile("test/TestChartValid.yaml")
+	missingFields := c.checkMandatoryFields()
+
+	if len(missingFields) > 0 {
+		t.Errorf("Test should pass since all mandatory fields are set.")
+	}
+}
+
+func TestShouldFailIfMandatoryFieldsIsMissing(t *testing.T) {
```

Comment:
Same as before. Test names should be documentation and therefore describe business requriements.
Here: what kind of fields are mandatory? Of course listing every field in the test name is not a good solution, but maybe we can find a grouping name

Reply:
Function renamed to TestShouldFailIfConfigurationSettingsAreMissing.